app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt

```diff
@@ -37,26 +40,30 @@ class SearchLocationFragment : Fragment() {
             thisActivity.supportActionBar?.title = ""
             thisActivity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
         }
-        rootView.locationProgressBar.visibility = View.GONE
+        setHasOptionsMenu(true)
 
         fromSearchFragment = arguments?.getBoolean(FROM_SEARCH) ?: false
 
-        if (thisActivity is Activity) geoLocationUI.configure(thisActivity, rootView, searchLocationViewModel)
+        geoLocationViewModel.currentLocationVisibility.observe(this, Observer {
+            rootView.currentLocation.visibility = View.GONE
+        })
+
+        geoLocationViewModel.configure(thisActivity)
+
+        rootView.currentLocation.setOnClickListener {
+            checkLocationPermission()
+            geoLocationViewModel.configure(thisActivity)
+            rootView.locationProgressBar.visibility = View.VISIBLE
+            geoLocationViewModel.location.observe(this, Observer { location ->
+                searchLocationViewModel.saveSearch(location)
+                redirectToMain()
+            })
```

Comment:
Move outside of click listener

Reply:
if I move redirect to main outside click listener with the observer, After opening location fragment again, value for current location fragment is already assigned. And redirectToMain() called instantly. So, user not able to change the location again.

Reply:
Then make the location a single event live data

Reply:
Same result with `SingeLiveEvent`. I think because of configure method is called at start. And location permission is already granted.